frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -185,6 +185,7 @@ function ReaderBookmark:onToggleBookmark()
     self.ui:handleEvent(Event:new("SetDogearVisibility",
                                   not self.view.dogear_visible))
     UIManager:setDirty(self.view.dialog, "ui")
+    UIManager:broadcastEvent(Event:new("BookmarksChanged"))
```

Comment:
Can't you re-use the already sent event SetDogearVisibility and rename your handler to `ReaderFooter:onSetDogearVisibility()` ?
(Also, any reason the existing one use self.ui:handleEvent and yours UIManager:broadcastEvent ?)

Reply:
> Can't you re-use the already sent event SetDogearVisibility and rename your handler to `ReaderFooter:onSetDogearVisibility()` ?

No, that does not work.

> (Also, any reason the existing one use self.ui:handleEvent and yours UIManager:broadcastEvent ?)

No, I have changed it to `self.ui:handleEvent(Event:new("BookmarksChanged"))`

Reply:
> No, that does not work.

I'm curious as to why/how it does not work. With your change, these should work the same:
```
self.ui:handleEvent(Event:new("SetDogearVisibility",...)
self.ui:handleEvent(Event:new("BookmarksChanged"))
```
Probably just because of the `return true` in readerdogear.lua's `ReaderDogear:onSetDogearVisibility(visible)` - that I guess you could remove so the event can propagate to other handlers.